src/AmbientSounds.Uwp/Controls/SettingsControl.xaml.cs

```diff
@@ -38,5 +45,45 @@ private void OnThemeSelectionChanged(object sender, SelectionChangedEventArgs e)
                 ViewModel.UpdateTheme(s);
             }
         }
+        
+        // Check if auto-startup is on and adjust its settings accordingly.
+        private async void AutoStartup_Toggled(object sender, Windows.UI.Xaml.RoutedEventArgs e)
+        {
+            StartupTask startupTask = await StartupTask.GetAsync("AmbieAutoStartup");
+            Debug.WriteLine(startupTask.State.ToString());
+            if ((sender as ToggleSwitch).IsOn)
+            {
+                switch (startupTask.State)
+                {
+                    case StartupTaskState.Disabled:
+                        StartupTaskState newState = await startupTask.RequestEnableAsync();
+                        if (newState == StartupTaskState.Enabled || newState == StartupTaskState.EnabledByPolicy)
+                        {
+                            (sender as ToggleSwitch).IsOn = true;
+                        }
+                        else { 
+                            (sender as ToggleSwitch).IsOn = false;
+                        }
```

Comment:
```suggestion
                        ((ToggleSwitch)sender).IsOn = newState == StartupTaskState.Enabled || newState == StartupTaskState.EnabledByPolicy;
```

Reply:
Honestly, I had no idea this could be done this way. Looks way cleaner, thanks.

Reply:
IsOn is a boolean property and you're checking for a boolean in the if statement, so using the boolean directly is the same, just cleaner.